hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -426,7 +429,9 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		}
 		removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolValidMachineConfigConditionType)
 
-		kubevirtBootImage, err = kubevirt.GetImage(nodePool, releaseImage)
+		cacheAnnotation, foundAnnotation := hcluster.GetAnnotations()[hyperv1.KubeVirtEnablerImageCachingAnnotation]
+		cachingEnabled := !foundAnnotation || cacheAnnotation == "true"
+		kubevirtBootImage, err = kubevirt.GetImage(nodePool, releaseImage, hostedClusterNS, log, cachingEnabled)
```

Comment:
it's a bit odd to pass in the logger here. Typically i see the ctx passed in, and a new logger derived from the ctx.

Reply:
Good point. However, if you'll look inside, this will only work if the context contains the logger in its `value`.

This will work here because the operatorSDK, I think, adds the logger to the context. This is not true for unit tests, though. so the test need to add a logger to the context.